hypex/comparators/abstract.py

```diff
@@ -125,41 +130,224 @@ def _extract_dataset(
+        data: List[Tuple[str, Dataset]]
+    ) -> List[Tuple[str, Dataset]]:
+        result = [
+            (f"{bucket[0]}{NAME_BORDER_SYMBOL}{column}", bucket[1][column])
+            for bucket in data
+            for column in bucket[1].columns
+        ]
+        return result
+
+    @classmethod
+    def _split_data_to_buckets(
+        cls,
+        data: Dataset,
+        compare_by: Literal["groups", "columns", "columns_in_groups", "cross"],
+        target_fields: Union[str, List[str]],
+        baseline_field: Optional[str] = None,
+        group_field: Optional[str] = None,
+    ) -> Tuple:
+        """
+        Splits the given dataset into buckets into baseline and compared data, based on the specified comparison mode.
+
+        Args:
+            data (Dataset): The dataset to be split.
+            group_field (Union[Sequence[str], str]): The field(s) to group the data by.
+            target_fields (Union[str, List[str]]): The field(s) to target for comparison.
+            compare_by (Literal['groups', 'columns', 'columns_in_groups', 'cross'], optional): The method to compare the data. Defaults to 'groups'.
+            baseline_field (Optional[str], optional): The column to use as the baseline for comparison. Required if `compare_by` is 'columns' or 'columns_in_groups'. Defaults to None.
+
+        Returns:
+            Tuple: A tuple containing the baseline data and the compared data.
+
+        Raises:
+            NoRequiredArgumentError: If `baseline_field` is None and `compare_by` is 'columns' or 'columns_in_groups' or 'cross'.
+            ValueError: If `compare_by` is not one of the allowed values.
+        """
+
+        target_fields = Adapter.to_list(target_fields)
+        if compare_by == "groups":
+            baseline_data, compared_data = cls._split_for_groups_mode(
+                data, group_field, target_fields
+            )
+        elif compare_by == "columns":
+            baseline_data, compared_data = cls._split_for_columns_mode(
+                data, baseline_field, target_fields
+            )
+        elif compare_by == "columns_in_groups":
+            baseline_data, compared_data = cls._split_for_columns_in_groups_mode(
+                data, group_field, baseline_field, target_fields
+            )
+        elif compare_by == "cross":
+            baseline_data, compared_data = cls._split_for_cross_mode(
+                data, group_field, baseline_field, target_fields
+            )
+        else:
+            raise ValueError(
+                f"Wrong compare_by argument passed {compare_by}. It can be only one of the following modes: 'groups', 'columns', 'columns_in_groups', 'cross'."
+            )
+        return baseline_data, compared_data
+
+    @classmethod
+    def _split_for_groups_mode(
+        cls, data: Dataset, group_field: str, target_fields: List[str]
+    ):
+        if isinstance(target_fields, List) and len(target_fields) > 1:
+            raise ValueError(
```

Comment:
May be you should use warrning instead of error? You can get first target from list andd do estimation.